mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmCreateVaultScreen.tsx

```diff
@@ -15,51 +15,84 @@ import { TextRow } from '@components/TextRow'
 import BigNumber from 'bignumber.js'
 import { NumberRow } from '@components/NumberRow'
 import { FeeInfoRow } from '@components/FeeInfoRow'
+import { ConversionTag } from '@components/ConversionTag'
+import { ConversionParam } from '@screens/AppNavigator/screens/Balances/BalancesNavigator'
+import { NativeLoggingProps, useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { WhaleWalletAccount } from '@defichain/whale-api-wallet'
+import { CTransactionSegWit } from '@defichain/jellyfish-transaction'
+import { onTransactionBroadcast } from '@api/transaction/transaction_commands'
+import { useWalletContext } from '@shared-contexts/WalletContext'
+import { fetchVaults } from '@store/loans'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
 
 type Props = StackScreenProps<LoanParamList, 'ConfirmCreateVaultScreen'>
 
 export function ConfirmCreateVaultScreen ({ route, navigation }: Props): JSX.Element {
-  const { loanScheme, fee } = route.params
+  const { loanScheme, fee, conversion } = route.params
+  const logger = useLogger()
   const hasPendingJob = useSelector((state: RootState) => hasTxQueued(state.transactionQueue))
   const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+  const currentBroadcastJob = useSelector((state: RootState) => firstTransactionSelector(state.ocean))
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [isOnPage, setIsOnPage] = useState<boolean>(true)
+  const { address } = useWalletContext()
+  const client = useWhaleApiClient()
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    setIsOnPage(true)
+    return () => {
+      setIsOnPage(false)
+    }
+  }, [])
 
   function onCancel (): void {
-    navigation.navigate({
-      name: 'CreateVaultScreen',
-      params: {
-        loanScheme
-      },
-      merge: true
-    })
+    if (!isSubmitting) {
```

Comment:
For some reason `isSubmitting` is never reflecting the most updated value when user submits (stay false forever), the `hasPendingJob` and `hasBroadcastJob` flag is already blocking the cancel button through `disabled` props. 

So i suggest we can remove this flag entirely

Reply:
updated